src/backend/env/mod.rs

```diff
@@ -1624,33 +1660,20 @@ impl State {
         mutate(|state| {
             state.distribution_reports.clear();
             state.distribute_realm_revenue(now);
-        });
-
-        // We only mint and distribute if no open proposals exists
-        if read(|state| state.proposals.iter().all(|p| p.status != Status::Open)) {
-            mutate(|state| {
-                state.minting_mode = true;
-                state.mint();
-                state.minting_mode = false;
-            });
-            State::distribute_icp().await;
-            mutate(|state| {
-                for summary in &state.distribution_reports {
-                    state.logger.info(format!(
-                        "{}: {} [[details](#/distribution)]",
-                        summary.title, summary.description
-                    ));
-                }
-            });
-        } else {
-            mutate(|state| {
-                state
-                    .logger
-                    .info("Skipping minting & distributions due to open proposals")
-            });
-        }
 
+            state.minting_mode = true;
+            state.mint();
+            state.minting_mode = false;
```

Comment:
Why do we set the minting mode to true then false?

Reply:
The minting logic checks deeps inside for this flag being set. It's just one countermeasure to avoid triggering minting (which is nothing else but a transfer from an anonymous principal) from unexpected places.